run.py

```diff
@@ -262,94 +263,111 @@ def iss_cmp(test_list, iss, output_dir, isa, verbose):
-parser.add_argument("--verbose", dest="verbose", action="store_true",
-                    help="Verbose logging")
-parser.add_argument("--co", dest="co", action="store_true",
-                    help="Compile the generator only")
-parser.add_argument("--so", dest="so", action="store_true",
-                    help="Simulate the generator only")
-parser.add_argument("--cmp_opts", type=str, default="",
-                    help="Compile options for the generator")
-parser.add_argument("--sim_opts", type=str, default="",
-                    help="Simulation options for the generator")
-parser.add_argument("--steps", type=str, default="all",
-                    help="Run steps: gen,gcc_compile,iss_sim,iss_cmp")
-parser.add_argument("--lsf_cmd", type=str, default="",
-                    help="LSF command. Run in local sequentially if lsf \
-                          command is not specified")
-parser.add_argument("--gen_timeout", type=int, default=360,
-                    help="Generator timeout limit in seconds")
-parser.add_argument("--iss_timeout", type=int, default=50,
-                    help="ISS sim timeout limit in seconds")
-
-parser.set_defaults(co=False)
-parser.set_defaults(so=False)
-parser.set_defaults(verbose=False)
-
-args = parser.parse_args()
-cwd = os.path.dirname(os.path.realpath(__file__))
-
-if not args.iss_yaml:
-  args.iss_yaml = cwd + "/yaml/iss.yaml"
-
-if not args.simulator_yaml:
-  args.simulator_yaml = cwd + "/yaml/simulator.yaml"
-
-if not args.testlist:
-  args.testlist = cwd + "/yaml/testlist.yaml"
-
-# Create output directory
-subprocess.run(["mkdir", "-p", args.o])
-subprocess.run(["mkdir", "-p", ("%s/asm_tests" % args.o)])
-
-# Process regression test list
-matched_list = []
-process_regression_list(args.testlist, args.test, args.iterations, matched_list)
-if len(matched_list) == 0:
-  sys.exit("Cannot find %s in %s" % (args.test, args.testlist))
-
-# Run instruction generator
-if args.steps == "all" or re.match("gen", args.steps):
-  gen(matched_list, args.simulator, args.simulator_yaml, args.o,
-      args.so, args.co, args.lsf_cmd, args.seed, cwd,
-      args.cmp_opts, args.sim_opts, args.gen_timeout, args.verbose)
-
-if not args.co:
-  # Compile the assembly program to ELF, convert to plain binary
-  if args.steps == "all" or re.match("gcc_compile", args.steps):
-    gcc_compile(matched_list, args.o, args.isa, args.mabi, args.verbose)
-
-  # Run ISS simulation
-  if args.steps == "all" or re.match("iss_sim", args.steps):
-    iss_sim(matched_list, args.o, args.iss, args.iss_yaml,
-            args.isa, args.iss_timeout, args.verbose)
-
-  # Compare ISS simulation result
-  if args.steps == "all" or re.match("iss_cmp", args.steps):
-    iss_cmp(matched_list, args.iss, args.o, args.isa, args.verbose)
+  Returns: The created parser.
+  """
+  # Parse input arguments
+  parser = argparse.ArgumentParser()
+
+  parser.add_argument("--o", type=str,
+                      help="Output directory name")
```

Comment:
I think this will trigger an error is user doesn't pass "--o xxx"

Reply:
For default, `output_dir = "out_" + str(date.today())`

Reply:
I mean --o becomes a mandatory argument now, there's no way user can run this script without specifying --o.

Reply:
I can run this script without specifying --o.
```
danghai@ubuntu:~/google/riscv-dv$ python3 run.py --test riscv_arithmetic_basic_test --simulator questa
Processing regression test list : /home/danghai/google/riscv-dv/yaml/testlist.yaml, test: riscv_arithmetic_basic_test
Found matched tests: riscv_arithmetic_basic_test, iterations:2
Processing simulator setup file : /home/danghai/google/riscv-dv/yaml/simulator.yaml
Found matching simulator: questa
Building RISC-V instruction generator
Running RISC-V instruction generator
Generating 2 riscv_arithmetic_basic_test
```

Output: `out_2019-08-03`